packages/extension-webgazer/src/index.ts

```diff
@@ -0,0 +1,296 @@
+import { JsPsych, JsPsychExtension, JsPsychExtensionParameters } from "jspsych";
+
+// we have to add webgazer to the global window object because webgazer attaches itself to
+// the window when it loads
+declare global {
+  interface Window { webgazer: any; }
+}
+
+interface InitializeParameters extends JsPsychExtensionParameters {
+  round_predictions: boolean;
+  auto_initialize: boolean;
+  sampling_interval: number;
+  webgazer: any;
+}
+
+interface OnStartParameters extends JsPsychExtensionParameters {
+  targets: Array<string>;
+}
+
+class WebGazerExtension implements JsPsychExtension {
+
+  constructor(private jsPsych: JsPsych) { }
+
+  // private state for the extension
+  // extension authors can define public functions to interact
+  // with the state. recommend not exposing state directly
+  // so that state manipulations are checked.
+  private currentTrialData = [];
+  private currentTrialTargets = {};
+  private currentTrialSelectors;
+  private domObserver;
+  private webgazer;
+  private initialized = false;
+  private currentTrialStart;
+  private activeTrial = false;
+  private sampling_interval;
+  private round_predictions;
+  private gazeInterval;
+  private gazeUpdateCallbacks;
+  private currentGaze;
```

Comment:
If you already know the type of something here, add it with `:` – otherwise, TypeScript uses `any`.